### ISSUE TITLE
Escaping double quotes in EMBL/GenBank feature qualifiers

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#1689]
Quoting http://www.insdc.org/files/feature_table.html#3.3

> # 3.3.3.1 Free text
> Most qualifier values will be a descriptive text phrase which must be enclosed 
> in double quotation marks. When the text occupies more than one line, a single 
> set of quotation marks is required at the beginning and at the end of the 
> text. The text itself may be composed of any printable characters (ASCII 
> values 32-126 decimal). If double quotation marks are used within a free text 
> string, each set (") must be 'escaped' by placing a second double quotation 
> mark immediately before it (""). For example: 
> ```
>               /note="This is an example of ""escaped"" quotation marks"
> ```

Currently we strip out all double quotes here on parsing:
https://github.com/biopython/biopython/blob/biopython-171/Bio/GenBank/__init__.py#L1159

And on writing just wrap as given with double quotes:
https://github.com/biopython/biopython/blob/biopython-171/Bio/SeqIO/InsdcIO.py#L313

Implementing this as per the specification should not be too complicated, but it will need new tests added.

See also the related handling of value-less qualifiers, #905

**Comment by Gasta88** [comment]
Hi @peterjc, I would like to give it a go and solve this issue. I have a couple of questions before:
1) Is `Bio.GenBank` supposed to be deprecated in favor of `Bio.SeqIO`? If yes, should the editing being made on this last module instead of `Bio.GenBank`?

2) For testing the solution for this issue, does a new `.gb` file have to be created inside `Tests\GenBank` or is there another way preferred?

3) Is it correct too assume that for the "writing" part, only the user is in charge of quoting internally and the module takes care of the external double quotes? If that is so, `Bio.SeqIO.InsdcIO` might not need some editing.

**Comment by peterjc** [comment]
1. No, ``Bio.GenBank`` is not deprecated - it is used for parsing by ``Bio.SeqIO``, and also offers a non-``SeqRecord`` based representation which is closer to the file format structure (and can be used for writing GenBank files).

2. Rather than yet another test file, you can follow the examples of existing tests which construct a record from code (or load a known good file, and then modify the record in code). Then you can check output to GenBank format (and that the quotes are escaped) and then re-parse that output (and check the quotes are un-escaped on loading).

3. The end user should not be doing anything special with the quotes (other than the standard Python syntax to include a double quote within a string - the simplest way being wrapping it with single quotes), we should try to handle things like this:

```
my_feature.qualifiers["note"] = ['This is a "silly" example']
my_feature.qualifiers["note"] = ['This has two double quotes in a row, "", as a test case!']
my_feature.qualifiers["note"] = ['"To be, or not to be?"']
```

**Comment by peterjc** [comment]
Fixed via #1818, thank you @alonalj
